File: php/ESInterface.php
<?php
class ESInterface {
        static $url='';
        static function setURL($str)
        {
                $lastoff=strlen($str)-1;
                if($str[$lastoff]=='/'){
                        rtrim($str, "/");
                }
                self::$url=$str;
        }
        static function getSingle($index, $type, $id)
        {
                return ESInterface::curlOps(self::$url."/$index/$type/".strval($id), 'GET');
        }
        static function getRange($index, $offset=0, $size=10)
        {
                return ESInterface::curlOps(self::$url."/$index/_search?from=".strval($offset)."&size=".strval($size), 'GET');
        }
        static function searchFilter($index, $type, $filters, $sort="", $offset=0, $size=10)
        {
                //$data=array("query"=> array("filtered"=> array("filter"=> array("bool"=> array("must"=> $filters)))));
                $data=array("query"=> array("bool"=> array("filter"=> $filters)));
                if($sort!=""){
                        $sort="sort=$sort&";
                }
                return ESInterface::curlOps(self::$url."/$index/$type/_search?".$sort."from=".strval($offset)."&size=".strval($size), 'GET', $data);
        }
        static function searchQuery($index, $type, $queries, $sort="", $offset=0, $size=10)
        {
                $data=array("query"=> array("bool"=> array("must"=> $queries)));
                if($sort!=""){
                        $sort="sort=$sort&";
                }
                return ESInterface::curlOps(self::$url."/$index/$type/_search?".$sort."from=".strval($offset)."&size=".strval($size), 'GET', $data);
        }
        static function searchMixed($index, $type, $filters, $queries, $sort="", $from=0, $size=10)
        {
                $data=array("query"=> array("filtered"=> array("filter"=> array("bool"=> array("must"=> $filters)),"query"=> array("bool"=> array("must"=> $queries)))));
                if($sort!=""){
                        $sort="sort=$sort&";
                }
                return ESInterface::curlOps(self::$url."/$index/$type/_search?".$sort."from=".strval($from)."&size=".strval($size), 'POST', $data);
        }
        static function curlOps($url, $op, $data=NULL)
        {
                $curl=curl_init($url);
                curl_setopt($curl, CURLOPT_CUSTOMREQUEST, $op);
                curl_setopt($curl, CURLOPT_HEADER, false);
                curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
                curl_setopt($curl, CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
                if($data!=NULL) {
                        curl_setopt($curl, CURLOPT_POSTFIELDS, json_encode($data));
                }
                $response=curl_exec($curl);
                return $response;
        }

        static function deleteIndex($index)
        {
                ESInterface::curlOps(self::$url."/$index", 'DELETE');
        }

        static function createIndex($index)
        {
                ESInterface::curlOps(self::$url."/$index", 'PUT');
        }

        static function setMapping($index, $type, $field, $declArray)
        {
                $rURL=self::$url."/$index/$type/_mapping";
                return self::curlOps($rURL, 'PUT', array($type => array('properties' => array($field => $declArray))));
        }
        static function indexSingle($index, $type, $id, $data)
        {
                $rURL=self::$url."/$index/$type/$id";
                return ESInterface::curlOps($rURL, 'PUT', $data);
        }
        static function deleteSingle($index, $type, $id)
        {
                $rURL=self::$url."/$index/$type/$id";
                return ESInterface::curlOps($rURL, 'DELETE');
        }
};
?>

File: python/ESInterface.py
#!/usr/bin/env python

import json, requests

class ESInterface:
        """ElasticSearch client"""
        url=""
        ver=5
        headers = {'content-type': 'application/json', 'Accept-Charset': 'UTF-8'}

        def __init__(self):
                pass

        def setURL(self, url):
                if url.endswith('/'):
                        url=url[:-1]
                self.url=url

        def setVersion(self, ver):
                self.ver=ver

        def curlOps(self, url, op, data=None):
                if data is None:
                        r=requests.request(op, url, headers=self.headers)
                else:
                        r=requests.request(op, url, data=json.dumps(data), headers=self.headers)
                return r.text

        def getSingle(self, _index, _type, _id):
                return self.curlOps(self.url+"/"+_index+"/"+_type+"/"+str(_id), 'GET')

        def searchFilter(self, _index, _type, _filters, _sort='', _offset=0, _size=10):
                if self.ver<5:
                        data={'query':{'filtered':{'filter':{'bool':{'must': _filters}}}}}
                else:
                        data={'query':{'bool':{'filter': _filters}}}
                if _sort!='':
                        _sort="sort="+_sort+"&"
                return self.curlOps(self.url+"/"+_index+"/"+_type+"/_search?"+_sort+"from="+str(_offset)+"&size="+str(_size), 'GET', data)

        def searchQuery(self, _index, _type, _queries, _sort='', _offset=0, _size=10):
                data={'query':{'bool':{'must': _queries}}}
                if _sort!='':
                        _sort="sort="+_sort+"&"
                return self.curlOps(self.url+"/"+_index+"/"+_type+"/_search?"+_sort+"from="+str(_offset)+"&size="+str(_size), 'GET', data)

        def searchMixed(self, _index, _type, _filters, _queries, _sort='', _offset=0, _size=10):
                if self.ver<5:
                        data={'query':{'filtered':{'filter':{'bool':{'must':_filters}}, 'query': {'bool': {'must': _queries}}}}}
                else:
                        data={'query':{'bool':{'must': _queries, 'filter': _filters}}}
                if _sort!='':
                        _sort="sort="+_sort+"&"
                return self.curlOps(self.url+"/"+_index+"/"+_type+"/_search?"+_sort+"from="+str(_offset)+"&size="+str(_size), 'POST', data)

        def deleteIndex(self, _index):
                self.curlOps(self.url+"/"+_index, 'DELETE')

        def createIndex(self, _index):
                self.curlOps(self.url+"/"+_index, 'PUT')

        def setMapping(self, _index, _type, _field, _declArray):
                rURL=self.url+"/"+_index+"/"+_type+"/_mapping";
                return self.curlOps(rURL, 'PUT', {_type: {'properties':{_field: _declArray}}})

        def indexSingle(self, _type, _id, _data):
                rURL=self.url+"/"+_index+"/"+_type+"/"+str(_id)
                return self.curlOps(rURL, 'PUT', _data)

        def deleteSingle(self, _index, _type, _id):
                rURL=self.url+"/"+_index+"/"+_type+"/"+str(_id)
                return self.curlOps(uURL, 'DELETE')
